### PR TITLE
Secure compass with JWT and OAuth

### PR DESCRIPTION
**Pull request by kubadz**
**Description**

Changes proposed in this pull request:

- use Oauthkeeper to secure compass endpoints

**Related issue(s)**

### CODE REVIEW

File: resources/compass/charts/gateway/templates/oathkeeper-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+apiVersion: oathkeeper.ory.sh/v1alpha1
+kind: Rule
+metadata:
+  name: compass-gateway-oauth
+  namespace: compass-system
```
Comment (pkosiec) [inline resources/compass/charts/gateway/templates/oathkeeper-rules.yaml]: Please remove namespace

File: resources/compass/charts/gateway/templates/oathkeeper-rules.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        url: {{ .Values.global.tenantMappingService.url }}:{{ .Values.global.tenantMappingService.port }}
+  - handler: id_token
+    config:
+      claims: {{ .Values.oathkeeper.idTokenConfig.claims | quote }}
+---
+apiVersion: oathkeeper.ory.sh/v1alpha1
+kind: Rule
+metadata:
+  name: compass-gateway-jwt
+  namespace: compass-system
```
Comment (pkosiec) [inline resources/compass/charts/gateway/templates/oathkeeper-rules.yaml]: Please remove namespace

File: resources/compass/charts/gateway/values.yaml [hunk 3]
```diff
@@ -9,5 +9,11 @@ oathkeeper:
   host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
   port: 4455
 
+oathkeeper:
+  host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
+  port: 4455
+  idTokenConfig:
+    claims: "{\"scope\": \"{{ print .Extra.scopes }}\", \"tenants\": \"{{ print .Extra.tenants }}\"}"
```
Comment (pkosiec) [inline resources/compass/charts/gateway/values.yaml]:
```suggestion
    claims: "{\"scopes\": \"{{ print .Extra.scope }}\", \"tenant\": \"{{ print .Extra.tenant }}\"}"
```

File: resources/compass/charts/gateway/values.yaml [hunk 4]
```diff
@@ -9,5 +9,11 @@ oathkeeper:
   host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
   port: 4455
 
+oathkeeper:
+  host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
```
Comment (pkosiec) [inline resources/compass/charts/gateway/values.yaml]: `kyma-system` shouldn't be hardcoded, there should be `Release.namespace` inserted, right?

Comment (kubadz) [reply resources/compass/charts/gateway/values.yaml]: `Release.namespace` won't work, as ory components are installed as a part of ory chart, which in Kyma is installed in `kyma-system` namespace.

File: resources/compass/values.yaml [hunk 5]
```diff
@@ -89,3 +94,7 @@ global:
       host: "localhost"
       hostPort: "5432"
       sslMode: ""
+
+  tenantMappingService:
+    port: 3000
+    url: http://compass-healthchecker.compass-system.svc.cluster.local
```
Comment (pkosiec) [inline resources/compass/values.yaml]:
```suggestion
    url: http://compass-director.compass-system.svc.cluster.local:3001/tenant-mapping
```

File: resources/compass/values.yaml [hunk 6]
```diff
@@ -15,8 +15,8 @@ global:
       dir: develop/
       version: "1ce3b6c3"
     healthchecker:
-      dir: develop/
-      version: "26601526"
+      dir: pr/
+      version: "PR-287"
```
Comment (pkosiec) [inline resources/compass/values.yaml]: Please reset change for healthchecker image. Unfortunately we cannot have `PR-XXX` images here.

**Comment by pkosiec** [comment]
Please also add `ory` into `installer-cr-cluster-compass.yaml.tpl` file. Thanks!

File: resources/compass/values.yaml [hunk 7]
```diff
@@ -89,3 +94,7 @@ global:
       host: "localhost"
       hostPort: "5432"
       sslMode: ""
+
+  tenantMappingService:
+    port: 3000
```
Comment (pkosiec) [inline resources/compass/values.yaml]: The port is now unnecessary, right?

Comment (pkosiec) [inline resources/compass/values.yaml]: The port is now unnecessary, right? Please remove it

File: resources/compass/charts/gateway/templates/oathkeeper-rules.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    url: <http|https>://{{ .Values.global.gateway.tls.secure.oauth.host }}.{{ .Values.global.ingress.domainName }}/<.*>
+  authenticators:
+  - handler: oauth2_introspection
+  authorizer:
+    handler: allow
+  mutators:
+  - handler: hydrator
+    config:
+      api:
+        url: {{ .Values.global.tenantMappingService.url }}:{{ .Values.global.tenantMappingService.port }}
```
Comment (pkosiec) [inline resources/compass/charts/gateway/templates/oathkeeper-rules.yaml]:
```suggestion
        url: {{ .Values.global.tenantMappingService.url }}
```

File: resources/compass/charts/gateway/templates/oathkeeper-rules.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  - handler: jwt
+    config:
+      trusted_issuers: ["https://dex.{{ .Values.global.ingress.domainName }}"]
+  authorizer:
+    handler: allow
+  mutators:
+  - handler: hydrator
+    config:
+      api:
+        url: {{ .Values.global.tenantMappingService.url }}:{{ .Values.global.tenantMappingService.port }}
```
Comment (pkosiec) [inline resources/compass/charts/gateway/templates/oathkeeper-rules.yaml]:
```suggestion
        url: {{ .Values.global.tenantMappingService.url }}
```

**Pull request merged**